packages/mermaid/src/docs/community/development.md

```diff
@@ -1,77 +1,151 @@
+### 1. Checkout a git branch
 
-**One example:**
+Mermaid uses a [Git Flow](https://guides.github.com/introduction/flow/)–inspired approach to branching.
 
-`feature/945_state_diagrams`
+Development is done in the `develop` branch.
 
-**Another example:**
+Once development is done we branch a `release` branch from `develop` for testing.
 
-`bug/123_nasty_bug_branch`
+Once the release happens we merge the `release` branch with `master` and delete the `release` branch. The live product and on-line documentation are what is in the `master` branch.
 
-## Contributing to Documentation
+**All new work should be based on the `develop` branch.**
 
-If it is not in the documentation, it's like it never happened. Wouldn't that be sad? With all the effort that was put into the feature?
+**When you are ready to do work, always, ALWAYS:**
 
-The docs are located in the `src/docs` folder and are written in Markdown. Just pick the right section and start typing. If you want to propose changes to the structure of the documentation, such as adding a new section or a new file you do that via **[.vitepress/config.ts](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/.vitepress/config.ts)**.
+1. Make sure you have the most up to date version of the `develop` branch. (fetch or pull to update it)
+2. Check out the `develop` branch
+3. Create a new branch for your work. Please name the branch following our naming convention below.
 
-> **All the documents displayed in the GitHub.io page are listed in [.vitepress/config.ts](https://github.com/mermaid-js/mermaid/blob/develop/packages/mermaid/src/docs/.vitepress/config.ts)**.
+We use the follow naming convention for branches:
 
-The contents of [https://mermaid-js.github.io/mermaid/](https://mermaid-js.github.io/mermaid/) are based on the docs from the `master` branch. Updates committed to the `master` branch are reflected in the [Mermaid Docs](https://mermaid-js.github.io/mermaid/) once released.
+```text
+   [feature | bug | chore | docs]/[issue number]_[short description using dashes ('-') or underscores ('_') instead of spaces]
+```
 
-## How to Contribute to Documentation
+- The first part is the **type** of change: a feature, bug, chore, or documentation change ('docs')
+- followed by a _slash_ (which helps to group like types together in many git tools)
+- followed by the **issue number**
+- followed by an _underscore_ ('\_')
+- followed by a short text description (but use dashes ('-') or underscores ('\_') instead of spaces)
 
-We are a little less strict here, it is OK to commit directly in the `develop` branch if you are a collaborator.
+If your work is specific to a single diagram type, it is a good idea to put the diagram type at the start of the dscription. This will help use keep release notes organized: it will help us keep changes for a diagram type together.
```

Comment:
Looks like cSpell isn't working since it didn't catch it.  I even ran the lint script and it didn't catch it.

(Separately - I think we should revised the `package.json` scripts for documentation: things have settled down a bit and it looks like there's some duplication.)

(Also separately:  I'd like to see the Issue templates changed so that people could check off which diagram(s) their issue pertains to. Would be nice if this could automate some issue labeling as well.  No idea if this is possible or feasible. Will eventually post an issue about this.)

Reply:
> Wording seems a bit off/duplicated?

Agree!  My brain is saturated.  Suggestions welcomed. :-)